packages/plugin-ma-tracking/src/ma-tracking.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  AdditionalBrowserOptions,
+  BaseEvent,
+  BeforePlugin,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import type { AmplitudeBrowser } from '@amplitude/analytics-browser';
+import { AdvancedCampaignTracker } from './advanced-campaign-tracker';
+
+export class MATrackingPlugin implements BeforePlugin {
+  name = 'MATracking';
+  type = PluginType.BEFORE as const;
+
+  constructor(private readonly instance: AmplitudeBrowser, private readonly options: AdditionalBrowserOptions = {}) {}
```

Comment:
If this only works in browser we should add a postfix to the plugin name to specify `plugin-marketing-analytics-browser`